handlers/vault.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/gaia-pipeline/gaia/security"
+
+	"github.com/gaia-pipeline/gaia"
+	"github.com/labstack/echo"
+)
+
+type secret struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+// AddSecret creates a secret using the vault.
+func AddSecret(c echo.Context) error {
+	s := new(secret)
+	err := c.Bind(s)
+	if err != nil {
+		gaia.Cfg.Logger.Error("error reading secret", "error", err.Error())
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	v, err := security.NewVault()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error initializing vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	v.Add(s.Key, []byte(s.Value))
```

Comment:
I'm missing an open here so this is overwriting whatever is in the vault at the moment. Which brings me to rename the functions to make it more apparent how this needs to be used.